core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala

```diff
@@ -0,0 +1,145 @@
+package org.broadinstitute.dsde.rawls.entities.datarepo
+
+import com.google.cloud.bigquery.Field.Mode
+import com.google.cloud.bigquery._
+import org.broadinstitute.dsde.rawls.entities.exceptions.{DataEntityException, EntityNotFoundException}
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+ * contains helper methods for working with BigQuery from an EntityProvider.
+ */
+trait DataRepoBigQuerySupport {
+
+  /**
+   * translate a single BigQuery FieldValue to a single Rawls AttributeValue
+   *
+   * @param field the BQ schema's field definition
+   * @param fv the BQ field value
+   * @return the Rawls attribute value, post-translation
+   */
+  def fieldValueToAttributeValue(field: Field, fv: FieldValue): AttributeValue = {
+    field.getType match {
+      case x if fv.isNull =>
+        AttributeNull
+      case LegacySQLTypeName.FLOAT | LegacySQLTypeName.INTEGER | LegacySQLTypeName.NUMERIC =>
+        AttributeNumber(fv.getNumericValue)
+      case LegacySQLTypeName.BOOLEAN =>
+        AttributeBoolean(fv.getBooleanValue)
+      case LegacySQLTypeName.STRING =>
+        AttributeString(fv.getStringValue)
+      case LegacySQLTypeName.RECORD =>
+        // TODO: unclear what to do with RECORD types; they don't translate cleanly to the entity model
+        AttributeString(fv.getValue.toString)
+      case _ =>
+        // "else" case that covers LegacySQLTypeNames of:
+        // DATE, DATETIME, TIME, TIMESTAMP, BYTES, GEOGRAPHY
+        // these types don't have strongly-typed equivalents in the entity model, so we treat them
+        // as string values, and rely on the caller to parse the string
+        AttributeString(fv.getValue.toString)
+    }
+  }
+
+  /**
+   * Given a BigQuery row and one named field from the BQ schema, return a Rawls AttributeName->Attribute pair
+   * representing that field's value(s) inside the row.
+   *
+   * @param field the BQ schema's field definition
+   * @param row all field values for a single BQ row
+   * @return the Rawls attributename->attribute pair, post-translation
+   */
+  def fieldToAttribute(field: Field, row: FieldValueList): (AttributeName, Attribute) = {
+    val attrName = field.getName
+    val fv = Try(row.get(attrName)) match {
+      case Success(fieldValue) => fieldValue
+      case Failure(ex) =>
+        throw new DataEntityException(s"field ${field.getName} not found in row ${asDelimitedString(row)}")
+    }
+
+    val attribute = field.getMode match {
+      case Mode.REPEATED =>
+        fv.getRepeatedValue.asScala.toList match {
+          case x if x.isEmpty => AttributeValueEmptyList
+          case members =>
+            // can a list itself contain a list? Recursion here is difficult because of the Attribute/AttributeValue hierarchy
```

Comment:
I am not well-versed in BigQuery (so hopefully this is even the correct documentation) but it looks like arrays of arrays are not supported per https://cloud.google.com/bigquery/docs/reference/standard-sql/arrays